### PR TITLE
Support for pagination in v2 engine of `SELECT * FROM <table>` queries

### PR DESCRIPTION
**Pull request by MaxKsyunz**
### Description
v2 SQL engine can now paginate simple queries. Pagination is initiated by setting `fetch_size` property in the request JSON.

For example, initiate a pagination request, send:
```
curl -s -XPOST http://localhost:9200/_plugins/_sql -H 'Content-Type: application/json' -d '{"query": "SELECT * from calcs", "fetch_size": 5}'
```

Send the following to get a subsequent page:
```
curl -s -XPOST http://localhost:9200/_plugins/_sql -H 'Content-Type: application/json' -d '{"cursor": "...." }'
```

Each response to a pagination request will include a `cursor` property if there is more data available. The last page in the paginated request will not have a `cursor` property.

Pagination is implemented using the [OpenSearch Scroll API](https://opensearch.org/docs/latest/api-reference/scroll/). Please see [pagination-v2.md](https://github.com/opensearch-project/sql/blob/9ebdda5f315acdfeef473e38512efcb5eedc5905/docs/dev/Pagination-v2.md) for implementation details.
 
### Check List
- [x] New functionality includes testing.
  - [x] All tests pass, including unit test, integration test and doctest
- [x] New functionality has been documented.
  - [x] New functionality has javadoc added
  - [x] New functionality has user manual doc added
- [x] Commits are signed per the DCO using --signoff 

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

**Review by acarbonetto — APPROVED** [review]

**Review by acarbonetto — DISMISSED** [review]

File: opensearch/src/main/java/org/opensearch/sql/opensearch/storage/scan/OpenSearchIndexScanBuilder.java [hunk 1]
```diff
@@ -24,36 +24,39 @@
  * by delegated builder internally. This is to avoid conditional check of different push down logic
  * for non-aggregate and aggregate query everywhere.
  */
-public class OpenSearchIndexScanBuilder extends TableScanBuilder {
+public abstract class OpenSearchIndexScanBuilder extends TableScanBuilder {
```
Comment (penghuo) [inline opensearch/src/main/java/org/opensearch/sql/opensearch/storage/scan/OpenSearchIndexScanBuilder.java:27]: follow up on same question on https://github.com/opensearch-project/sql/pull/1600.
why it is abstract and no subclass?

File: opensearch/src/main/java/org/opensearch/sql/opensearch/storage/scan/OpenSearchIndexScanBuilder.java [hunk 2]
```diff
@@ -24,36 +24,39 @@
...
-  public OpenSearchIndexScanBuilder(OpenSearchIndexScan indexScan) {
-    this.delegate = new OpenSearchIndexScanQueryBuilder(indexScan);
+  protected OpenSearchIndexScanBuilder(PushDownQueryBuilder translator) {
+    this.delegate = translator;
   }
 
   @Override
   public TableScanOperator build() {
-    return delegate.build();
+    return createScan(delegate.build());
```
Comment (penghuo) [inline opensearch/src/main/java/org/opensearch/sql/opensearch/storage/scan/OpenSearchIndexScanBuilder.java:55]: what is the difference of build and createScan?

Comment (MaxKsyunz) [reply opensearch/src/main/java/org/opensearch/sql/opensearch/storage/scan/OpenSearchIndexScanBuilder.java:55]: `OpenSearchIndexScanBuilder.createScan` -- create an OpenSearchIndexScan based on the provided  OpenSearchRequest. 

`OpenSearchIndexScanBuilder.build` -- build an OpenSearchRequest and an OpenSearchIndexScan based on it.

They will be merged as part of making `OpenSearchIndexScanBuilder` inner class of `OpenSearchIndexScan`. 

Related to the [other](https://github.com/opensearch-project/sql/pull/1666#discussion_r1210810444) discussion about builders.

**Review by acarbonetto — DISMISSED** [review]

**Review by GumpacG — APPROVED** [review]

**Review by Yury-Fridlyand — APPROVED** [review]
Observation: there are about 15 files with no-op changes - whitespaces or comment updating. Pagination feature had bigger scope, but after few rounds of reworks and improvements, some changes were reverted, but not comments.
I also found few typos in javadocs, nothing critical. Will be fixed in the following work on Pagination feature.

No issues nor objections found while code review and manual testing!

**Review by forestmvey — APPROVED** [review]

**Pull request merged**